### DISCUSSION TITLE
Disable pruning for some resources

### DISCUSSION THREAD

**Discussion by jlewi** [fluxcd/flux2#763] [Q&A] [answered]
Is it possible to override the pruning behavior for individual resources?

I'd like to configure my KustomizationRepository to do pruning but I have a small subset of resources that I would like Flux to stop managing as opposed to delete in the event they are removed from the repo. The use case is to protect resources for which deletion would be destructive and guard against accidental removal from the repo.

Examples

* PV/PVCs
* Namespaces

Use case

* Use Flux to create a namespace and some other resources
* Someone creates resources in that namespace outside of flux
* Someone deletes the manifests in the Flux repo (including the Namespace resource)
* I want Flux to stop managing the Namespace as opposed to deleting it
* Flux should prune other resources it was managing

**Comment by stefanprodan** [comment, marked as answer]
Place the "protected" resources in a separate directory in Git and create a Flux Kustomization for it with `prune: false`.

**Reply by jlewi** [reply]
It looks like Flux now supports this with annotations.
https://toolkit.fluxcd.io/components/kustomize/kustomization/#garbage-collection
